src/libs/ajax.js

```diff
@@ -1012,6 +1033,73 @@ const Buckets = signal => ({
 
       getObject,
 
+      rename: async newName => {
+        await copy(newName, bucket, false)
+        return doDelete()
+      }
+    }
+  },
+
+  analysis: (namespace, bucket, name, analysisType) => {
+    const bucketUrl = `storage/v1/b/${bucket}/o`
+
+    const encodeFileName = n => Utils.switchCase(analysisType,
+      ['python', () => nbName(getDisplayName(n))], ['r', () => rName(getDisplayName(n))])
+
+    const calhounPath = () => Utils.switchCase(analysisType,
+      ['python', () => 'api/convert'], ['r', () => 'api/convert/rmd'])
+
+    const mimeType = () => Utils.switchCase(analysisType, ['python', () => 'application/x-ipynb+json'], ['r', () => 'text/plain'])
```

Comment:
Is there a reason for these to be functions that are called rather than values that are just evaluated?

Reply:
ah yeah, I think they originally were somewhere else, doesn't make sense with this refactor though 👍